### ISSUE TITLE
dependency clashes on new installs (possibly mkl?)

### ISSUE THREAD

**Issue by michaelaye** [conda/conda#2141]
I see currently some dependency clashes that ruin the usefulness of `conda update --all` for me (I guess for everybody?).

It's something to do with numpy to mkl relationship, but possibly more packages are involved, or it's simply the dependency calculator of conda that is confused?

Here's how to reproduce:

`conda create -n py35 python=3 numpy`

Output:
```bash
Fetching package metadata: ....
Solving package specifications: .............
Package plan for installation in environment /Users/klay6683/miniconda3/envs/py35:

The following NEW packages will be INSTALLED:

    mkl:        11.3.1-0
    numpy:      1.10.4-py35_0
    openssl:    1.0.2f-0
    pip:        8.0.3-py35_0
    python:     3.5.1-0
    readline:   6.2-2
    setuptools: 20.1.1-py35_0
    sqlite:     3.9.2-0
    tk:         8.5.18-0
    wheel:      0.29.0-py35_0
    xz:         5.0.5-1
    zlib:       1.2.8-0

Proceed ([y]/n)? y

Linking packages ...
[      COMPLETE      ]|#########################################################################| 100%
#
# To activate this environment, use:
# $ source activate py35
#
# To deactivate this environment, use:
# $ source deactivate
#
```
Then I change into the new environment and do an update --all:
```bash
$ py35act
discarding /Users/klay6683/miniconda3/bin from PATH
prepending /Users/klay6683/miniconda3/envs/py35/bin to PATH
(py35)
klay6683 at macd2860 in ~
$ conda update --all
Fetching package metadata: ....
Solving package specifications: .....................
Package plan for installation in environment /Users/klay6683/miniconda3/envs/py35:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    mkl-rt-11.1                |               p0        66.6 MB
    numpy-1.10.2               |          py35_p0         2.6 MB
    ------------------------------------------------------------
                                           Total:        69.2 MB

The following NEW packages will be INSTALLED:

    mkl-rt: 11.1-p0

The following packages will be DOWNGRADED:

    numpy:  1.10.4-py35_0 --> 1.10.2-py35_p0 [mkl]

```
When I repeat the same experiment with install numpy, scipy and scikit-learn, conda wants to downgrade scipy drastically to 0.16.1 from 0.17:
```bash
$ conda update --all
Fetching package metadata: ....
Solving package specifications: .........................
Package plan for installation in environment /Users/klay6683/miniconda3/envs/py35:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    scipy-0.16.1               |     np110py35_p0        10.7 MB
    scikit-learn-0.17          |     np110py35_p1         3.6 MB
    ------------------------------------------------------------
                                           Total:        14.3 MB

The following NEW packages will be INSTALLED:

    mkl-rt:       11.1-p0

The following packages will be DOWNGRADED:

    numpy:        1.10.4-py35_0      --> 1.10.2-py35_p0      [mkl]
    scikit-learn: 0.17.1-np110py35_0 --> 0.17-np110py35_p1   [mkl]
    scipy:        0.17.0-np110py35_0 --> 0.16.1-np110py35_p0 [mkl]

Proceed ([y]/n)? n
```
Is there a workaround for this that keeps `update --all` useful or do I have to check each package manually now for existing updates?

**Comment by jakirkham** [comment]
Install `nomkl` in the environment. Remove `mkl` and `mkl-rt`. Try to update again.

**Comment by michaelaye** [comment]
hold on, do you mean that only the right-hand versions are mkl-boosted? That's why they are older, b/c they did not yet compile the new ones with mkl?

**Comment by jakirkham** [comment]
I believe the ones that were in the environment previously are linked against Accelerate (system BLAS) on Mac, OpenBLAS on Linux, and MKL on Windows. In short, on Windows you should get the same result in either case.

Going forward Mac and Linux will have the same linkages, but now track the `nomkl` package. An `mkl` package has been added to those platforms so that you can use MKL linked versions. On Windows, there is no option to not use MKL so its version will appear as tracking `mkl`.

The `mkl-rt` package I don't believe is supported any more. It is getting pulled in by accident. Not sure why this happens, but have seen this also. This package is the reason you are getting old versions `numpy`, `scipy`, and `scikit-learn`. Removing it should be sufficient for getting back to the newer versions of those packages and keeping MKL if you wish.

**Comment by michaelaye** [comment]
Can't be it, as I don't have `mkl-rt` installed:
```bash
$ conda remove mkl-rt
Fetching package metadata: ......
Error: no packages found to remove from environment: /Users/klay6683/miniconda3/envs/py35
```

**Comment by jakirkham** [comment]
Ah, sorry missed that you say no to installing that package. Honestly, there are some things that need to be sorted with `mkl` I would simply recommend using `nomkl`.

**Comment by asmeurer** [comment]
This is a dupe of https://github.com/conda/conda/issues/2062.

**Comment by jakirkham** [comment]
If that is the case maybe, an updated title and description on #2062 would help make that clear to people who are struggling with this surprising new problem. It seemed to me at first glance that the issue was geared towards whether the warning should be remove or if we need to pay for MKL valid but different concerns. It is true the conversation went away from that, but still making this clear from the beginning would help.

**Comment by asmeurer** [comment]
Done.

**Comment by jakirkham** [comment]
Thanks.

**Comment by kalefranz** [comment]
Should be fixed with `>= 4.0` releases.
